### PR TITLE
Bot stuff

### PR DESCRIPTION
**Pull request by slipher**
Add some functionality to the bot AI to avoid getting stuck so often. Plus a couple other small bugs.

### CODE REVIEW

**Comment by illwieckz** [comment]
From what I see they all managed to escape from there:

bot trap

but, it looks like they fall in another issue:

bot trap
bot trap

**Comment by illwieckz** [comment]
interesting bot stuck case:

bot trap

**Comment by illwieckz** [comment]
Perhaps it's related to another bug (we already seen similar things in the past), but it looks like these bots are trying to _not walk on navmeshes_.

bot trap

**Comment by illwieckz** [comment]
Something that can be good to keep in mind is blocks can have navmeshes in them:
bot trap
bot trap

I don't know if it can lead to some bugs, it looks like these navmeshes are too fare from outside to “grab” a bot, by the way.

**Comment by illwieckz** [comment]
Well, I'm not able to reproduce on other maps like plat23 for example, looks to be an another and unrelated bug.

**Comment by illwieckz** [comment]
That other bug is related to the fact the eggs are placed in disconnected navmeshes. Definitely another bug. So everything looks OK to me in this PR.

**Comment by illwieckz** [comment]
Example in some maps:

bot escape

Well this one was very unlucky, but found a way.
bot escape

This one can't escape with current code (it has too much place to move), these are the 10%:
bot escape
By the way he felt while evolving to dragoon.

**Comment by illwieckz** [comment]
As noted on the IRC channel, the current "give bp" implementation can overflow, it must be fixed before merging.

And it's time to ask a question, it would be the time to give to "build points" a proper name, since they are now mined as a resource, it's more than a "point", it's part of the unvanquished universe.

I remember the other "give" cheat available: ammo, fuel, funds, health, momentum, poison, stamina. So, how can be named build ressources?

**Comment by slipher** [comment]
Sometimes the problems are caused by navmesh defects. Here is one where you can a polygon going through a wall:
image

**Comment by slipher** [comment]
Regarding `give bp`, I see using cheats as similar to "voiding your warranty" for the duration of the game — it's the user's fault if they break it with overflow. Besides, none of the other code that adds BP checks for overflow, so it seems incongruous to check it in only this one place.

**Comment by illwieckz** [comment]
> Besides, none of the other code that adds BP checks for overflow.

Ok, so it means it's another issue elsewhere and is probably not preventing the merge, see #987 for the dedicated thread about that specific bug which is not related to stuck bots.

**Comment by illwieckz** [comment]
For some reason, I prefer to merge the engine split first even if this PR does not modify any engine files, because this way the last commit before the split commit on this repository is also the last commit before the split commit in the other repository. It will help future historiographer work.

As wrote there:
https://github.com/Unvanquished/Unvanquished/pull/983#issuecomment-279776063
I will merge the split in two days. I guess we will be able to merge the present PR right after if no one reviewer as something to say against merging that bot stuff.

File: src/sgame/sg_cmds.cpp [hunk 1]
```diff
@@ -575,6 +575,12 @@ void Cmd_Give_f( gentity_t *ent )
 		G_AddMomentumGeneric( (team_t) ent->client->pers.team, amount );
 	}
 
+	if ( Q_strnicmp( name, "bp", strlen("bp") ) == 0 )
+	{
+		float bp = trap_Argc() < 3 ? 300.0f : atof( name + strlen("bp") );
```
Comment (DolceTriade) [inline src/sgame/sg_cmds.cpp]: Why not just reuse amount?

**Comment by DolceTriade** [comment]
Looks good. i don't feel strongly about the amount vs bp change.

File: src/sgame/sg_cmds.cpp (hunk 1 above)
Comment (slipher) [inline src/sgame/sg_cmds.cpp]: As I understand, more narrowly scoped variables are usually preferable. `amount` is only there at the top because of C89.

**Comment by illwieckz** [comment]
so, is this can be merged?

**Pull request merged**